src/awkward/_do.py

```diff
@@ -304,61 +304,92 @@ def mergemany(contents: list[Content]) -> Content:
 def reduce(
     layout: Content,
     reducer: ak._reducers.Reducer,
-    axis: int = -1,
+    axis: int | None = -1,
```

Comment:
Is there a type for `AxisMaybeNone` that could be used here? (And what happens if an `axis` is passed an `np.int64`? We should accept it.)